### PR TITLE
EDU-746 Remove public rooms

### PR DESCRIPTION
**Pull request by adrianaluchian**
Closes https://educandu.atlassian.net/browse/EDU-746

### CODE REVIEW

File: src/api-clients/room-api-client.js [hunk 1]
```diff
@@ -51,7 +50,7 @@ class RoomApiClient {
   deleteAllPrivateRoomsForUser({ ownerId }) {
```
Comment (ahelmberger) [inline src/api-clients/room-api-client.js:50]: We could even rename it to `deleteRoomsForUser`

File: src/components/pages/users.less [hunk 2]
```diff
@@ -1,26 +1,11 @@
...
-
-  .UsersPage-roleTag--superUser {
-    color: @red-10;
-  }
-
-  .UsersPage-roleTag--editor {}
-
-  .UsersPage-roleTag--superEditor {}
-}
-
```
Comment (ahelmberger) [inline src/components/pages/users.less:15]: Ha :rofl:

File: src/api-clients/room-api-client.js (hunk 1 above)
Comment (adrianaluchian) [reply src/api-clients/room-api-client.js]: Course, forgot the client-side, I had renamed just server-side 😵‍💫 
Thanks, done!

File: src/components/pages/users.less (hunk 2 above)
Comment (adrianaluchian) [reply src/components/pages/users.less:15]: Exactly my reaction 😂

File: src/server/document-controller.js [hunk 3]
```diff
@@ -84,8 +80,8 @@ class DocumentController {
 
     const room = doc.roomId ? await this.roomService.getRoomById(doc.roomId) : null;
 
-    if (room?.access === ROOM_ACCESS.private && !isRoomOwnerOrMember({ room, userId: user._id })) {
-      throw new Forbidden();
+    if (!!room && !isRoomOwnerOrMember({ room, userId: user._id })) {
+      throw new Unauthorized();
```
Comment (ahelmberger) [inline src/server/document-controller.js:84]: I think this was correct before. `Unauthorized` (401, bad naming for "unauthenticated") will simply redirect the user to the login page (and therefore into an endless loop), while we want `Forbidden` (403) :thinking:

Comment (adrianaluchian) [reply src/server/document-controller.js:84]: Ah so this isn't actually `Unauthorized` 😱 Would have been better than `Forbidden`.. Damn, ok, will revert all these places where I changed it.

Comment (adrianaluchian) [reply src/server/document-controller.js]: Reverted. Again what learned, thanks!

Comment (ahelmberger) [reply src/server/document-controller.js]: Yeah, not optimal naming from the early days of the web :-) Thanks!

**Comment by ahelmberger** [comment]
Such a joy to see a lot of conditions vanish :smile:

**Review by ahelmberger — APPROVED** [review]

**Comment by adrianaluchian** [comment]
> Such a joy to see a lot of conditions vanish 😄

Yes! Glad we're able to simplify before we add some more complexity 😬

**Pull request merged**